xapian-letor/configure.ac

```diff
@@ -385,6 +385,12 @@ AH_BOTTOM(
 ])
 
 AC_DEFINE([XAPIAN_LIB_BUILD], [1], [Workaround for avoiding sub-header included error])
+AC_DEFINE([BUILD_BACKEND_GLASS], [1], [Workaround for avoiding setting the conditionals])
+AC_DEFINE([BUILD_BACKEND_INMEMORY], [1], [Workaround for avoiding setting the conditionals])
+AC_DEFINE([BUILD_BACKEND_REMOTE], [1], [Workaround for avoiding setting the conditionals])
+AM_CONDITIONAL([BUILD_BACKEND_GLASS], true)
+AM_CONDITIONAL([BUILD_BACKEND_INMEMORY], true)
+AM_CONDITIONAL([BUILD_BACKEND_REMOTE], true)
```

Comment:
I made the mistake of not changing the Makefile.mk in letor/harness while implementing `#ifdef` ..... I did rectify the mistake and it now works perfectly.